### PR TITLE
PACA-960: Update to RDA API 0.12.0

### PR DESCRIPTION
**Pull request by KyRobbins**
**JIRA Ticket:**
[PACA-960](https://jira.cms.gov/browse/PACA-960)

**User Story or Bug Summary:**

---

### What Does This PR Do?

This PR consumes the new fields from RDA for fiss revenue lines, etc., storing their values in the database.  This PR does **not** include serving this data through the FHIR API.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-model/bfd-model-rda/mappings/FissRevenueLine.yaml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        billed HCPCS code in the Outpatient Code Editor (OCE) module.'
+    - name: acoRedRarc
+      sqlType: varchar(5)
+      dbName: aco_red_rarc
+      comment: 'If the claim has a demonstration code equal to “59” (Pioneer ACO
+       Demonstration) or “68” (Pioneer ACO SNF), and provided service is from an outpatient
+        provider (TOB equal to 12, 13, 14, 22, 23, 34, 71, 72, 74, 75, 76, 77, 81, 82, 85)
+         or the service is for home health Part B; this field will identify the Remittance
+          Advice Remark Code (RARC) returned to the provider advising this decision and
+           possible appeal rights.'
```
Comment (brianburton) [inline apps/bfd-model/bfd-model-rda/mappings/FissRevenueLine.yaml:103]: Some of these comments are indented strangely.  There are some below and before this one as well.

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V102__rda_0_12_0_updates.sql [hunk 2]
```diff
@@ -0,0 +1,34 @@
+-- As part of the 0.10 RDA API update, new raw text fields are added for select date fields and FISS diag_cd_2 is now nullable
+
```
Comment (brianburton) [inline apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V102__rda_0_12_0_updates.sql:2]: 0.12

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimGenerator.java [hunk 3]
```diff
@@ -190,8 +202,20 @@ private void addRandomFieldValues(FissClaim.Builder claim) {
...
         });
     optional("totalChargeAmount", () -> claim.setTotalChargeAmount(randomAmount()));
-    optional("recdDtCymd", () -> claim.setRecdDtCymd(randomDate()));
-    optional("currTranDtCymd", () -> claim.setCurrTranDtCymd(randomDate()));
+    optional(
+        "recdDtCymd",
+        () -> {
+          String date = randomDate();
+          claim.setRecdDtCymd(date);
+          claim.setRecdDtCymdText(date);
```
Comment (brianburton) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimGenerator.java:210]: In the real data stream is `recvd_dt_cymd` null and `recd_dt_cymd_text` populated when the text is an invalid date?  To keep the behavior that the two match on a valid date maybe use a `oneOf` for this with lambdas that set either just the text (invalid date case) or both (valid date case)?

Comment (KyRobbins) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimGenerator.java:210]: Looked into this, and it seems on previous fields where a `_text` value was added, both values are sent.

**Review by brianburton — DISMISSED** [review]
Nice work!

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V102__rda_0_12_0_updates.sql [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    dcn varchar(23) NOT NULL,
+    rda_position smallint NOT NULL,
+    non_bill_rev_code varchar(1) NOT NULL,
+    rev_cd varchar(4),
+    rev_units_billed integer,
+    rev_serv_unit_cnt integer,
+    serv_dt_cymd date,
+    serv_dt_cymd_text varchar(10),
+    hcpc_cd varchar(5),
+    hcpc_ind varchar(1),
```
Comment (mackec1) [inline apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V102__rda_0_12_0_updates.sql:22]: Curious on the use of varchar(1) (or (2) for that matter); generally the 'breakeven' point for varchar is (3).  Char(1) or char(2) is more efficient for processing AND storage.

**Review by mackec1 — APPROVED** [review]
Kyle, nice work, well-documented, and the protbuf stuff was easy to decipher/follow.

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V102__rda_0_12_0_updates.sql (hunk 4 above)
Comment (brianburton) [reply apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V102__rda_0_12_0_updates.sql:22]: According to the [postgresql documentation](https://www.postgresql.org/docs/current/datatype-character.html) there is no performance difference between char and varchar.  Also varchar allows empty strings, which char does not.

Of course this is assuming that AWS's postgresql uses the same storage layout for character data as postgresql.

**Review by brianburton — APPROVED** [review]

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V102__rda_0_12_0_updates.sql (hunk 4 above)
Comment (mackec1) [reply apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V102__rda_0_12_0_updates.sql:22]: brian, as always, interacting with you is a great learning experience; need to throw out some of my db knowlege from non-postgres db(s)! Ran the following example to prove to myself

```
drop table foo;
CREATE TEMP TABLE foo (id int, v_char char(1) not null, v_small varchar(1), v_big varchar(255));
insert into foo values(1, 'a', 'b', 'testing...123');

SELECT pg_column_size(id)        AS id
     , pg_column_size(v_char)    AS v_char
	 , pg_column_size(v_small)   AS v_small
     , pg_column_size(v_big)     AS v_big
     , pg_column_size(foo)       AS foo
FROM   foo
union all
SELECT pg_column_size(1)
     , pg_column_size('a'::char)
     , pg_column_size('b'::varchar)
     , pg_column_size('testing...123'::varchar)
     , pg_column_size(ROW(1, 'foo'::varchar, '12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890'::varchar));
```

Comment (brianburton) [reply apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V102__rda_0_12_0_updates.sql:22]: Nice experiment!  I just learned this recently myself so don't give me too much credit. :-)

**Pull request merged**